### PR TITLE
556 graph service - add node with extensible fields

### PR DESCRIPTION
**Pull request by doori**
- Add Node class with extensible fields
- Update graph service code to take all fields from the Node

### CODE REVIEW

File: data_processing/services/graph_service.py [hunk 1]
```diff
@@ -49,31 +56,42 @@ def update_graph(project, table, data_config_file, app_config_file):
...
 		# subset dataframe
-		pdf = df.select(F.col(src_column_name).alias(src), F.col(target_column_name).alias(target)) \
-			.groupBy(src, target) \
+		fields = src_node_fields + target_node_fields
+		# alias removes top-level column name if '.' is part of the column name: i.e. metadata.PatientName -> PatientName
+		src_alias = [x[x.find('.')+1:] for x in src_node_fields]
+		target_alias = [x[x.find('.')+1:] for x in target_node_fields]
+		fields_alias = src_alias + target_alias
+
+		pdf = df.select([F.col(c).alias(a) for c, a in zip(fields, fields_alias)]) \
```
Comment (aauker) [inline data_processing/services/graph_service.py:79]: Nice!

**Comment by aauker** [comment]
Great work, local tests were successful. The formalization of the different (sub) graphs is a great start. 
Only remarks for future work are:
1. The graph DB connection parameters historically have come from env, and I think this should continue.
2. I believe we can eventually adapt our general template schema to include a GRAPH option, in the case a graph integration is needed.  For now I suppose this will be passed via the -t argument.  
3. Separating the graph type and table path might also be needed given the new data model.

**Review by aauker — APPROVED** [review]
Approved.

**Comment by doori** [comment]
Thanks for the comments!
About the configuration, Arfath and I had some discussions, and we converged on havings some configs under the individual packages, for now. https://github.com/msk-mind/data-processing/pull/55#discussion_r534516559

**Pull request merged**